atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1058,6 +1059,26 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
         getMeter(AtlasDbMetricNames.SNAPSHOT_TRANSACTION_CELLS_RETURNED, tableRef).mark(resultCount.get());
     }
 
+    /**
+     * A sentinel becomes orphaned if the table has been truncated between the time where the write occurred
+     * and where it was truncated. In this case, there is a chance that we end up with a sentinel with no
+     * valid AtlasDB cell covering it. In this case, we ignore it.
+     */
+    private Set<Cell> findOrphanedSentinels(TableReference table, Map<Cell, Value> rawResults) {
+        Set<Cell> sweepSentinels = Maps.filterValues(rawResults, SnapshotTransaction::atDeletionSentinel).keySet();
+        if (sweepSentinels.isEmpty()) {
+            return Collections.emptySet();
+        }
+        Map<Cell, Long> atMaxTimestamp = keyValueService.getLatestTimestamps(
+                table,
+                Maps.asMap(sweepSentinels, x -> Long.MAX_VALUE));
+        return Maps.filterValues(atMaxTimestamp, ts -> Value.INVALID_VALUE_TIMESTAMP == ts).keySet();
+    }
+
+    private static boolean atDeletionSentinel(Value value) {
```

Comment:
would this be better described as the sweep sentinel? deletion sentinel seems overloaded with empty byte array that gets written when "deleting" a cell

Reply:
done